subprojects/common/http-client/src/main/kotlin/com/avito/http/internal/ServiceMetricsInterceptor.kt

```diff
@@ -0,0 +1,28 @@
+package com.avito.http.internal
+
+import okhttp3.Interceptor
+import okhttp3.Response
+import okio.IOException
+import java.net.SocketTimeoutException
+
+internal class ServiceMetricsInterceptor(private val listener: ServiceEventsListener) : Interceptor {
+
+    override fun intercept(chain: Interceptor.Chain): Response = try {
+        val response = chain.proceed(chain.request())
+
+        val latency = response.receivedResponseAtMillis - response.sentRequestAtMillis
+
+        listener.onResponse(
```

Comment:
We must distinguish requests to different methods.
It's important for troubleshooting.


- `api/upload` - 1k RPM, error rate 50%
- `api/fetch` - 10k RPM, error rate 0.1%

Without distinguishing we will see 11k RPM with error rate 0.4%. We would think that everything is ok.